src/main/java/uk/ac/cam/cl/dtg/isaac/api/AssignmentFacade.java

```diff
@@ -310,8 +311,25 @@ public Response getAssigned(@Context final HttpServletRequest request,
 
             if (null == groupIdOfInterest) {
                 List<UserGroupDTO> allGroupsOwnedAndManagedByUser = this.groupManager.getAllGroupsOwnedAndManagedByUser(currentlyLoggedInUser, false);
-                return Response.ok(this.assignmentManager.getAllAssignmentsForSpecificGroups(allGroupsOwnedAndManagedByUser, true))
-                        .cacheControl(getCacheControl(NEVER_CACHE_WITHOUT_ETAG_CHECK, false)).build();
+                List<AssignmentDTO> assignments = this.assignmentManager.getAllAssignmentsForSpecificGroups(allGroupsOwnedAndManagedByUser, true);
+                Set<String> gameboardIds = assignments.stream().map(AssignmentDTO::getGameboardId).collect(Collectors.toSet());
+                List<GameboardDTO> liteGameboards = this.gameManager.getLiteGameboards(gameboardIds);
+                // Remove unneeded data from the gameboards - very messy but not as messy as feasible alternatives
+                liteGameboards.forEach(g -> {
+                    g.setContents(null);
+                    g.setWildCard(null);
+                    g.setWildCardPosition(null);
+                    g.setGameFilter(null);
+                    g.setLastVisited(null);
+                    g.setPercentageCompleted(null);
+                    g.setCreationMethod(null);
+                    g.setCreationDate(null);
+                });
```

Comment:
OK, this works just fine, it's a bit messy but gets the job done and is explicit enough. However, given that we use mappers elsewhere to do a similar thing, wouldn't it be more idiomatic (I'm not saying "better" on purpose here :) to make a `LiteGameboardDTO` and use a mapper to get all the unnecessary data stripped?